pkg/executor/brie.go

```diff
@@ -99,11 +127,51 @@ func (p *brieTaskProgress) GetCurrent() int64 {
 func (p *brieTaskProgress) Close() {
 	p.lock.Lock()
 	current := atomic.LoadInt64(&p.current)
-	if current < p.total {
-		p.cmd = fmt.Sprintf("%s Canceled", p.cmd)
+	total := atomic.LoadInt64(&p.total)
+	cmd := p.cmd
+	if current < total && !strings.HasSuffix(p.cmd, " Canceled") {
+		cmd = fmt.Sprintf("%s Canceled", cmd)
+		p.cmd = cmd
 	}
-	atomic.StoreInt64(&p.current, p.total)
 	p.lock.Unlock()
+
+	if p.executor == nil {
+		log.Error("BRIE task executor is nil", zap.Uint64("task id", p.taskID))
+		return
+	}
+	updateMetaTable(context.Background(), p.executor, p.taskID, map[string]any{
+		"progress": current * 100 / total,
+		"state":    cmd,
+	})
+}
+
+func (p *brieTaskProgress) cleanup(e *exec.BaseExecutor, info *brieTaskInfo) {
+	p.lock.Lock()
+	cmd := p.cmd
+	current := atomic.LoadInt64(&p.current)
+	total := atomic.LoadInt64(&p.total)
+	//FIXME: Is this a good standard?
```

Comment:
How about add `Finish()` into the interface `br/pkg/glue/glue.go: Progress`

Reply:
Seems it will involve large scale modifications?